probe-rs/src/debug/mod.rs

```diff
@@ -221,16 +222,29 @@ impl<'debuginfo, 'probe, 'core> StackFrameIterator<'debuginfo, 'probe, 'core> {
     ) -> Self {
         let registers = Registers::from_core(core);
         let pc = address;
-
+        //Evaluate the static scoped variables.
+        let static_variables = match debug_info.get_stack_statics(core, pc) {
+            Ok(static_variables) => static_variables,
+            Err(err) => {
+                let mut error_variable = Variable::new();
+                error_variable.name = "ERROR".to_string();
+                error_variable.set_value(format!("Failed to retrieve static variables: {:?}", err));
+                vec![error_variable]
```

Comment:
I would prefer it if we use the usual Rust types for errors. Can we just store the actual error, instead of converting it into a String here?

Reply:
I chose to remain consistent with the idea of not failing when components of variable evaluation fails. This way, it filters into the UI of a DAP consumer like VSCode, and the user can access all the other variables, even if parts fail.   This is how most of the debugger errors are handled.   

Another alternative would be to exclude the particular element (frame, group, variable) and log an error to the Debug Console.   The reason I chose not to do this, is because as a user, I found it difficult to reconcile logged errors with specific variables I expected to see in the variable viewer. This solution allowed me to place the error in context.

I am open to negotiation/feedback on this approach, but it would require a fairly large rework on error handling in the debugger  ... enough scope for a new PR.

Reply:
I think this is problematic when probe-rs is used as a library. For another tool, it is impossible to detect if the variables are valid, or if an error occurred.

If you want to show the error message in the same place as the variable, I don't see a problem with that. But in my opinion, this should be done in the debugger itself, meaning in the debug adapter.

Reply:
I somewhat agree with you. In an ideal world, I would change the MS DAP protocol to allow for handling this in a way that allows closer to RUST pure code, and still maintain an optimum user experience. I always err on the side of better user experience. 

That said, our point about API use is totally valid.   My suggestion is that we merge this PR (the error handling via variable value predates the PR), and then I will open a new PR, and think about (and propose)  a way to get both RUST error's and a good user experience in VSCode. Sound fair?

Reply:
@Tiwalun ... rough outline. I am thinking that I might implement a flag that we pass in through a struct like `DebugInfo` (???), to leave the default behaviour as you suggest, and then modify it from the `debugger` app to get the behaviour I'm looking for. What do you think?

```
/// ErrorResult is a struct used to control how the API responds to errors. This allows the API to be used in a 'default' RUST way, where errors stop processing. Alternatively allow a `partial-success is Ok` approach and use a more UX friendly capture of the errors, while trying to complete the overall debug processing as successfully as possible. For instance, the API may fail to evaluate a single variable during DWARF decoding, but will continue to attempt processing all the other variables successfully.
#[derive(Debug, Clone, PartialEq)]
pub enum OnError {
    /// This is the default behaviour in RUST. Simply process the Err(original_error) into the Result<>
    ErrorResult,
    /// Return the error as a Ok(String), in addition to logging it, and continue processing wherever possible.
    Continue,    
}
impl Default for OnError {
    fn default() -> Self {
        Self::ErrorResult
    }
```

Reply:
My solution would be simpler. I would just not store the static variables in the `StackFrameIterator` at all. From the debug adapter, you can then just call the `DebugInfo::get_stack_statics` function. 

In the debug adapter, you can then map the returned error to a variable, like it is done now. This allows any caller to handle errors in their preferred way.

Reply:
@Tiwalun I love your suggestion for not storing the statics in the `StackFrameIterator`. It resolves the comment higher up and I will resubmit that in this PR once I have it implemented.

However, I don't see how it addresses the issue with me intercepting the errors into a string value. I used that strategy throughout all the local and register variables also. I would think that if we're going to change that, then we should do it throughout.

Reply:
@Tiwalun ... I've just submitted the update that removes static variables as you suggested. Not sure why I thought I needed to store it originally, but I appreciate the diligence in the code review. Thanks